backend/main.py

```
@@ -14,42 +18,251 @@
+            "status": "success"
         }
-    except requests.exceptions.RequestException as e:
-        raise HTTPException(status_code=500, detail=f"Hugging Face API error: {str(e)}")
+        
+    except HTTPException:
+        # Re-raise HTTP exceptions as-is
+        raise
+    except Exception as e:
+        logger.error(f"Unexpected error in process_input: {str(e)}")
+        logger.error(f"Traceback: {traceback.format_exc()}")
+        raise HTTPException(status_code=500, detail=f"Internal server error: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)